language/move-prover/stackless-bytecode-generator/src/reaching_def_analysis.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) The Libra Core Contributors
```

Comment:
It would be good to discuss what this analysis will be used for.  That will help in reasoning about the soundness of the dataflow equations implemented by this analysis.

Reply:
The purpose of this is minimizing the huge and unnecessary amount of temporaries introduced by the stackless bytecode, which are most of the time just aliases. Those contribute right now to low quality of execution traces, because values can't be tracked back to to user variables, and we do not print temporaries (`$n`) locals.  But I haven't written the transformation yet which evaluates the analysis result.

Also, the `BytecodeContext` contains similar data, and the purpose is to replace this by this more principled approach.

The analysis itself, 'reaching definitions', is a standard text book example for data flow analysis.